android/src/main/java/com/oney/WebRTCModule/WebRTCDevicesManager.java

```diff
@@ -0,0 +1,247 @@
+        if(isWiredHeadsetPlugged){
+            WritableMap params = this.createWritableMap(DeviceType.HEADSET, "Wired headset", DeviceKind.AUDIO_INPUT.getKind());
+            enumerateDevicesArray.pushMap(params);
+        }else {
+            WritableMap paramsBuiltIn = this.createWritableMap(DeviceType.BUILT_IN_MICROPHONE, "Built in microphone", DeviceKind.AUDIO_INPUT.getKind());
+            enumerateDevicesArray.pushMap(paramsBuiltIn);
+
+            WritableMap paramsBuiltInSpeaker = this.createWritableMap(DeviceType.BUILT_IN_MICROPHONE_SPEAKER, "Built in speaker microphone", DeviceKind.AUDIO_INPUT.getKind());
+            enumerateDevicesArray.pushMap(paramsBuiltInSpeaker);
+        }
+
+        boolean isBluetoothHeadsetPlugged = Arrays.stream(audioInputDevices).anyMatch(device -> device.getType() == AudioDeviceInfo.TYPE_BLUETOOTH_SCO);
+        if(isBluetoothHeadsetPlugged){
+            WritableMap params = this.createWritableMap(DeviceType.BLUETOOTH, "Bluetooth microphone", DeviceKind.AUDIO_INPUT.getKind());
+            enumerateDevicesArray.pushMap(params);
+        }
+    }
+
+    private void fillAudioOutputDevices(WritableArray enumerateDevicesArray){
+        AudioDeviceInfo[] audioOutputDevices = this.audioManager.getDevices(AudioManager.GET_DEVICES_OUTPUTS);
+
+        boolean isWiredHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(
+                device -> device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADSET ||
+                        device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADPHONES
+        );
+
+        WritableMap params = null;
+        // At Android, if the Wired headset is plugged, we are unable to choose to use the built-in microphone. We can only choose to use the bluetooth or the default.
+        if(isWiredHeadsetPlugged){
+            params = this.createWritableMap(DeviceType.HEADSET, "Wired headset", DeviceKind.AUDIO_OUTPUT.getKind());
+        }else {
+            params = this.createWritableMap(DeviceType.EARPIECE, "Phone Earpiece", DeviceKind.AUDIO_OUTPUT.getKind());
+        }
+        enumerateDevicesArray.pushMap(params);
+
+        //speaker
+        params = this.createWritableMap(DeviceType.SPEAKER, "Speaker", DeviceKind.AUDIO_OUTPUT.getKind());
+        enumerateDevicesArray.pushMap(params);
+
+        boolean isBluetoothHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(device -> device.getType() == AudioDeviceInfo.TYPE_BLUETOOTH_SCO);
+        if(isBluetoothHeadsetPlugged){
+            params = this.createWritableMap(DeviceType.BLUETOOTH, "Bluetooth", DeviceKind.AUDIO_OUTPUT.getKind());
+            enumerateDevicesArray.pushMap(params);
+        }
+    }
+
+    private WritableMap createWritableMap(DeviceType deviceType, String label, String kind){
+        WritableMap audioMap = Arguments.createMap();
+        audioMap.putString("deviceId", deviceType.toString());
```

Comment:
i guess alternatively you could've just had a single field—`deviceId`—that contained the audio route, rather than have _both_ `deviceId` and `audioRoute`. the argument for this is that the customer can't really do anything with `deviceId` _anyway_ for audio devices, right? they can only specify a route?

was the reason for not doing it this way that it would be weird for the `label` to change (for example, between "wired headset" and "earpiece") _without_ a corresponding change in `deviceId`?

Reply:
also, i noticed in https://github.com/daily-co/pluot-core/pull/4664/files that you're planning on only considering the facing mode when the customer specifies a different input camera.

so the same argument applies there as to the `audioRoute`: why not just encode the facing mode _as_ the `deviceId`?

Reply:
benefits of changing both audio and video to encode everything in `deviceId` is that:
* then the customer doesn't have to also look around at other fields like `audioRoute` and `facing` to understand if a certain device is _actually_ a different specifiable
* we could simplify back to a single `setInputDevices` method (or a special RN version of it), rather than having separate `switchCamera()` and `setAudioRoute()` methods

Reply:
For the camera I totally agree, we could change that and proceed this way if we think is better. I haven't changed that because for the camera the default behavior was already implemented by react-native-webrtc this way. That is the only reason.

Reply:
For the audio, we have `audioInputs` and `audioOutputs`, if I use the `audioRoute` as `deviceId`, both of them would have the same `deviceId`, which would be really weird. Or otherwise we would not return the `audiooutputs`, which also would be weird.

That is the reason why I have chosen to add a new field `audioRoute`, that is kind of the link, between the `audioinput` and `audiooutput`.

What do you think ?

Reply:
Refactor to use `setAudioDevice(deviceId)` instead of `setAudioRoute(audioRoute)`